src/main/java/com/iexec/common/task/TaskDescription.java

```diff
@@ -142,6 +144,11 @@ public static TaskDescription toTaskDescription(String chainTaskId,
                 chainDeal.getParams().getIexecInputFiles() != null) {
             inputFiles = chainDeal.getParams().getIexecInputFiles();
         }
+        Map<String, String> secrets = Map.of();
+        if (chainDeal.getParams() != null &&
```

Comment:
Testing whether `chainDeal.getParams()` is null there is pretty strange as we use it a bunch of times without any caution later - [like here](https://github.com/iExecBlockchainComputing/iexec-common/blob/6273701ec7bc749f55d6f7cbe5b3c873f27d2aed/src/main/java/com/iexec/common/task/TaskDescription.java#L159).
So we're sure it'll throw an NPE if `chainDeal.getParams()` is null. What about adding a single test at the beginning of the method so that we're sure this value is never null later? BTW, that's already what's done for `chainDeal`.

Reply:
It would be possible if `chainDeal.getParams()` was the only call which could throw a NPE.
NPE can be thrown by the following calls: `getBotSize()`, `getChainApp()`, `getChainCategory()`, `getChainDataset()`
It is not possible to do a single test at the beginning with those multiple calls.
**I will remove the unnecessary check and add a TODO.**

Later, I would like to investigate how data read from the chain are populated in Object instances.
It would be great to guarantee that `chainDeal.getParams()` can never be null.

What does everyone think of this suggestion ?

Reply:
Actually `chainDeal.getParams()` cannot be null. See how it is created in the deal [here](https://github.com/iExecBlockchainComputing/iexec-common/blob/63f66080c6ab269897a7d3bd125479b3b6b0eef1/src/main/java/com/iexec/common/chain/IexecHubAbstractService.java#L403). Cf [here](https://github.com/iExecBlockchainComputing/iexec-common/blob/63f66080c6ab269897a7d3bd125479b3b6b0eef1/src/main/java/com/iexec/common/chain/ChainDeal.java#L71).

Reply:
Also just for information, the deal params field onchain is not guaranteed to be non-null. So the object `DealParams` will be defined but the content of [iexecArgs](https://github.com/iExecBlockchainComputing/iexec-common/blob/63f66080c6ab269897a7d3bd125479b3b6b0eef1/src/main/java/com/iexec/common/chain/ChainDeal.java#L81) could potentially be null which is not necessarily a problem.

Reply:
I will take another look to improve my understanding of the code.
Thanks for the help.

Reply:
See https://github.com/iExecBlockchainComputing/iexec-common/pull/298/commits/f3b818be8003c042ebf381f70553e7290a43b315